homeassistant/components/sensor/postnl.py

```diff
@@ -0,0 +1,124 @@
+"""
+Sensor for PostNL packages.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.postnl/
+"""
+import logging
+from datetime import timedelta, datetime
+import re
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_USERNAME, CONF_PASSWORD, CONF_SCAN_INTERVAL,
+                                 ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['postnl_api==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+DEFAULT_NAME = 'postnl'
+ICON = 'mdi:package-variant-closed'
+ATTRIBUTION = 'Information provided by PostNL'
+SCAN_INTERVAL = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+    cv.time_period,
+})
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the PostNL platform."""
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    name = config.get(CONF_NAME)
+    update_interval = config.get(CONF_SCAN_INTERVAL)
+
+    from postnl_api import PostNL_API
+
+    try:
+        api = PostNL_API(username, password)
+
+    except Exception:
+        _LOGGER.exception("Can't connect to the PostNL webservice")
+        return
+
+    add_devices([PostNLSensor(api, name, update_interval)], True)
+
+
+class PostNLSensor(Entity):
+    """PostNL Sensor."""
+
+    def __init__(self, api, name, interval):
+        """Initialize the sensor."""
+        self.friendly_name = name
+        self._name = name
+        self._attributes = None
+        self._state = None
+
+        self._api = api
+
+        self.update = Throttle(interval)(self._update)
```

Comment:
You can use the throttle if you want to limit the update interval customization. Create a constant for this and use that together with the throttle, eg `MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)`. The throttle will only be active per entity though so multiple entities can still hit the api faster than the throttle allows.

Please rename `_update` to `update` and put the throttle as a decorator on the `update` method.

Reply:
Can you help me a little out with this one? `put the throttle as a decorator on the update method.`

Reply:
Also, you said that `SCAN_INTERVAL` is already part of the config scheme. Why not use this for the throttling? So refer to SCAN_INTERVAL instead of creating a custom constant called `MIN_TIME_BETWEEN_UPDATES`.

Reply:
```py
@Throttle(MIN_TIME_BETWEEN_UPDATES)
def update(self):
...
```

Reply:
You don't want the throttling time to be configurable. You should decide on the min time between updates. Users will be able to customize the update interval down to that time.

Reply:
Ok, thanks. Updated!
However, I am a bit lost how to implement the SCAN_INTERVAL and a default. The API can be queried every minute if you would like to, however I would suggest a default of every 15 minutes (or more) since this doesn't change that often. Any examples of other components who do something like this?

Reply:
Just define `SCAN_INTERVAL` as a constant with the appropriate time interval at the module level. You can search for `SCAN_INTERVAL` to find examples.